Assets/MRTK/Core/Utilities/Editor/Setup/MixedRealityToolkitPreserveSettings.cs

```diff
@@ -0,0 +1,79 @@
+            "  <assembly fullname = \"Microsoft.MixedReality.Toolkit.SDK\" preserve=\"all\"/> \n " +
+            "  <!-- Core systems --> \n" +
+            "  <assembly fullname = \"Microsoft.MixedReality.Toolkit.Services.BoundarySystem\" preserve=\"all\"/> \n" +
+            "  <assembly fullname = \"Microsoft.MixedReality.Toolkit.Services.CameraSystem\" preserve=\"all\"/> \n" +
+            "  <assembly fullname = \"Microsoft.MixedReality.Toolkit.Services.DiagnosticsSystem\" preserve=\"all\"/> \n" +
+            "  <assembly fullname = \"Microsoft.MixedReality.Toolkit.Services.InputSystem\" preserve=\"all\"/> \n" +
+            "  <assembly fullname = \"Microsoft.MixedReality.Toolkit.Services.SceneSystem\" preserve=\"all\"/> \n" +
+            "  <assembly fullname = \"Microsoft.MixedReality.Toolkit.Services.SpatialAwarenessSystem\" preserve=\"all\"/> \n" +
+            "  <assembly fullname = \"Microsoft.MixedReality.Toolkit.Services.TeleportSystem\" preserve=\"all\"/> \n" +
+            "  <!-- Data providers --> \n" +
+            "  <assembly fullname = \"Microsoft.MixedReality.Toolkit.Providers.LeapMotion\" preserve=\"all\"/> \n" +
+            "  <assembly fullname = \"Microsoft.MixedReality.Toolkit.Providers.OpenVR\" preserve=\"all\"/> \n" +
+            "  <assembly fullname = \"Microsoft.MixedReality.Toolkit.Providers.UnityAR\" preserve=\"all\"/> \n" +
+            "  <assembly fullname = \"Microsoft.MixedReality.Toolkit.Providers.WindowsMixedReality.Shared\" preserve=\"all\"/> \n" +
+            "  <assembly fullname = \"Microsoft.MixedReality.Toolkit.Providers.WindowsMixedReality\" preserve=\"all\"/> \n" +
+            "  <assembly fullname = \"Microsoft.MixedReality.Toolkit.Providers.XRSDK.WindowsMixedReality\" preserve=\"all\"/> \n" +
+            "  <assembly fullname = \"Microsoft.MixedReality.Toolkit.Providers.WindowsVoiceInput\" preserve=\"all\"/> \n" +
+            "  <assembly fullname = \"Microsoft.MixedReality.Toolkit.Providers.XRSDK\" preserve=\"all\"/> \n" +
+            "  <!-- Extension services --> \n" +
+            "  <assembly fullname = \"Microsoft.MixedReality.Toolkit.Extensions.HandPhysics\" preserve=\"all\"/> \n" +
+            "  <assembly fullname = \"Microsoft.MixedReality.Toolkit.Extensions.Tracking\" preserve=\"all\"/> \n" +
+            "  <assembly fullname = \"Microsoft.MixedReality.Toolkit.Extensions.SceneTransitionService\" preserve=\"all\"/> \n" +
+            "</linker>";
+
+        /// <summary>
+        /// Ensure that a link.xml file exists in the MixedRealityToolkit.Generated folder.
+        /// This file is used to control the Unity linker's byte code stripping of MRTK assemblies.
+        /// </summary>
+        public static void EnsureLinkXml()
+        {
+            string generatedFolder = MixedRealityToolkitFiles.MapRelativeFolderPathToAbsolutePath(MixedRealityToolkitModuleType.Generated, "");
```

Comment:
For cases when MRTK is installed via UPM, will this actually put the link.xml file in the Packages or PackageCache folder, instead of assets folder?

Reply:
It should be created in MixedRealityToolkit.Generated, in Assets.

Reply:
Just like custom profiles